Docs: Contribution guideline for merging pull requests (#38800)
grafana/grafana: contribute/merge-pull-request.md

> # Merge a pull request

Reviewer: Should we add a section about draft PRs? Is it OK to open a draft PR, should maintainers review drafts, etc

Reviewer: right I was looking for a section header somewhere that wasn't about merging that I could add this suggestion to :joy:

- Needs backporting?

## Before merge

Before actually merging a pull request there's a couple of things to take into consideration.

### Format the pull request title

Make sure that the pull request title is properly formatted according to `<Area>: <Summary>` and try to make the summary short and understandable for the community as a whole.

All commits in a pull request is squashed when merged and the pull request title will be the the default subject line of the squashed commit message. It's also used for [changelog/release notes](#include-in-changelog-and-release-notes).

See [formatting guidelines](create-pull-request.md#formatting-guidelines) for more information.

### Assign a milestone

It's recommended to add a milestone to every pull request. This makes it easier to track what changes did go into a certain release. Without this you're basically left with going through git commits which could be a lot harder.

There's also various tooling built that in some cases requires a pull request to be assigned a milestone, for example [generating changelog/release notes](#include-in-changelog-and-release-notes).

### Include in changelog and release notes?

> Grafana are generating its [changelog](https://github.com/grafana/grafana/blob/main/CHANGELOG.md) and [release notes](https://grafana.com/docs/grafana/latest/release-notes/) based on merged pull requests. Including changes in the changelog/release notes is very important to provide a somewhat complete picture of what changes a Grafana release actually includes.

Reviewer 2: ```suggestion
At Grafana we generate the [changelog](https://github.com/grafana/grafana/blob/main/CHANGELOG.md) and [release notes](https://grafana.com/docs/grafana/latest/release-notes/) based on merged pull requests. Including changes in the changelog/release notes is very important to provide a somewhat complete picture of what changes a Grafana release actually includes.
```
